pyteal/ast/abi/bool.py

```diff
@@ -0,0 +1,98 @@
+from typing import Union, cast, Sequence
+
+from ...types import TealType
+from ..expr import Expr
+from ..int import Int
+from ..bytes import Bytes
+from ..binaryexpr import GetBit
+from ..ternaryexpr import SetBit
+from .type import Type
+from .uint import NUM_BITS_IN_BYTE
+
+
+class Bool(Type):
+    def __init__(self) -> None:
+        super().__init__(TealType.uint64)
+
+    def new_instance(self) -> "Bool":
+        return Bool()
+
+    def has_same_type_as(self, other: Type) -> bool:
+        return type(other) is Bool
+
+    def is_dynamic(self) -> bool:
+        return False
+
+    def byte_length_static(self) -> int:
+        # Not completely accurate since up to 8 consecutive bools will fit into a single byte
+        return 1
+
+    def get(self) -> Expr:
+        return self.stored_value.load()
+
+    def set(self, value: Union[bool, Expr, "Bool"]) -> Expr:
```

Comment:
nit: there is a possibility that one can directly set a `TealType.uint64` inside scratch value, do you think it would cause an issue?

Reply:
Good point, we may want to validate that the type is only 0 or 1 🤔

Reply:
I will add a check for this similar to uint max values